metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py

```diff
@@ -923,20 +934,47 @@ def get_connectors_manifest(self) -> List[ConnectorManifest]:
                         config=self.config,
                         report=self.report,
                     ).connector_manifest
+                elif connector_manifest.config.get("connector.class").startswith(
+                    "io.debezium.connector"
+                ):
+                    connector_manifest = DebeziumSourceConnector(
+                        connector_manifest=connector_manifest, config=self.config
+                    ).connector_manifest
                 else:
-                    # Debezium Source Connector lineages
-                    try:
-                        connector_manifest = DebeziumSourceConnector(
-                            connector_manifest=connector_manifest, config=self.config
-                        ).connector_manifest
+                    # empty list of customized connector
+                    if not self.config.generic_connectors:
+                        logger.warning(
+                            f"Detected undefined connector {connector_manifest.name}, but user defined no customized connectors. Please refer to Kafka Connect ingestion recipe to define this customized connector."
```

Comment:
this error checking becomes redundant given the check below, once the default is changed to []